gemfirexd/client/src/main/java/com/pivotal/gemfirexd/internal/client/am/Statement.java

```diff
@@ -2715,6 +2715,11 @@ void parseSqlAndSetSqlModes(String sql) throws SqlException {
             firstToken = getStatementToken(sql, idx + 1);
           }
         }
+        if (firstToken.equalsIgnoreCase("set")) {
+          if (sql.toLowerCase().contains(" schema ")) {
+              this.connection_.setSchemaSql_ = sql;
+          }
+        }
 // GemStone changes END
```

Comment:
Yes. I saw the standard and will try to make it more accurate for not matching or missing any other syntax used.